app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlAdapter.kt

```diff
@@ -37,7 +40,28 @@ sealed class AdapterItem(@LayoutRes val viewType: Int) {
     data class TabHeader(val isPrivate: Boolean, val hasTabs: Boolean) : AdapterItem(TabHeaderViewHolder.LAYOUT_ID)
     data class TabItem(val tab: Tab) : AdapterItem(TabViewHolder.LAYOUT_ID) {
         override fun sameAs(other: AdapterItem) = other is TabItem && tab.sessionId == other.tab.sessionId
+
+        // Tell the adapter exactly what values have changed so it only has to draw those
+        override fun getChangePayload(newItem: AdapterItem): Any? {
+            (newItem as TabItem).let {
+                val shouldUpdateUrl = newItem.tab.url != this.tab.url
+                val shouldUpdateHostname = newItem.tab.hostname != this.tab.hostname
+                val shouldUpdateTitle = newItem.tab.title != this.tab.title
+                val shouldUpdateSelected = newItem.tab.selected != this.tab.selected
+                val shouldUpdateMediaState = newItem.tab.mediaState != this.tab.mediaState
+
+                return AdapterItemDiffCallback.TabChangePayload(
```

Comment:
Should we file a followup issue to include the favicon in this payload and update from session listener?